app/src/main/java/com/hover/stax/data/repository/AccountRepositoryImpl.kt

```diff
@@ -30,60 +30,40 @@ class AccountRepositoryImpl(val accountRepo: AccountRepo, val channelRepo: Chann
     override val fetchAccounts: Flow<List<Account>>
         get() = accountRepo.getAccounts()
 
-    override suspend fun createAccounts(channels: List<Channel>) {
-        val defaultAccount = accountRepo.getDefaultAccountAsync()
-        channels.mapIndexed { index, channel ->
-            createAccount(channel, -1, defaultAccount == null && index == 0)
-        }
-    }
-
-    override suspend fun createAccount(channel: Channel, simSubscriptionId: Int, isDefault: Boolean) {
+    override suspend fun createAccount(channel: Channel, subscriptionId: Int, isDefault: Boolean): Account {
         val accountName: String = if (getFetchAccountAction(channel.id) == null) channel.name else PLACEHOLDER //placeholder alias for easier identification later
-        val account = Account(
```

Comment:
This shouldn't use the default constructor. There is a constructor that takes a channel as the argument for a reason.